vscode-extension/src/extension.ts

```diff
@@ -906,3 +912,27 @@ async function shareAIConfig(
       });
   }
 }
+
+/**
+ * Selects a Python interpreter and initiates the dependency installation flow.
+ */
+async function initialize(
+  context: vscode.ExtensionContext,
+  outputChannel: vscode.LogOutputChannel
+) {
+  // Make sure Python API is activated
+  const pythonApi: PythonExtension = await PythonExtension.api();
+
+  await new Promise((resolve, reject) => {
+    vscode.commands
+    .executeCommand("python.setInterpreter")
+    .then(async () => {
+    // .then triggers on user selection or exiting interpreter selection.
+      resolve(undefined);
+    });
+  });
```

Comment:
`await vscode.commands.executeCommand("python.setInterpreter");` should do the same thing. Shouldn't need to do a new promise and seems weird to use both await and promise.then in an async function

```suggestion
  await vscode.commands.executeCommand("python.setInterpreter");
```

Reply:
thanks for the clarification, updated